src/tribler-core/tribler_core/modules/libtorrent/settings.py

```diff
@@ -0,0 +1,70 @@
+import re
+from typing import List, Optional
+
+from pydantic import validator
+
+from tribler_common.network_utils import NetworkUtils
+from tribler_core.config.tribler_config_section import TriblerConfigSection
+
+
+@validator('port', 'anon_listen_port')
+def validate_port_with_minus_one(v):
+    assert v is None or -1 <= v <= NetworkUtils.MAX_PORT, 'Port must be in range [-1..65535]'
+    return v
+
+
+class LibtorrentSettings(TriblerConfigSection):
+    enabled: bool = True
+    port: Optional[int] = None
+    proxy_type: int = 0
+    proxy_server: str = ':'
+    proxy_auth: str = ':'
+    max_connections_download: int = -1
+    max_download_rate: int = 0
+    max_upload_rate: int = 0
+    utp: bool = True
+    dht: bool = True
+    dht_readiness_timeout: int = 30
+
+    anon_listen_port: Optional[int] = None
+    anon_proxy_type = 0
+
+    anon_proxy_server_ip: str = '127.0.0.1'
+    anon_proxy_server_ports: List[str] = ['-1', '-1', '-1', '-1', '-1']
+    anon_proxy_auth: str = ''
+
+    _port_validator = validator('port', 'anon_listen_port', allow_reuse=True)(validate_port_with_minus_one)
+
+    @validator('anon_proxy_server_ip')
+    def validate_ip_address(cls, v):
+        pattern = r'^((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$'
```

Comment:
:sunglasses:

Reply:
I'd suggest replacing this method with:

```python
def validate_ip_address(cls, v):
    return v if ipaddress.IPv4Network(v) else None
```

This is slightly easier to read and also gives in-depth tracebacks:

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "E:\Python38\lib\ipaddress.py", line 1449, in __init__
    self.network_address = IPv4Address(addr)
  File "E:\Python38\lib\ipaddress.py", line 1252, in __init__
    self._ip = self._ip_int_from_string(addr_str)
  File "E:\Python38\lib\ipaddress.py", line 1149, in _ip_int_from_string
    raise AddressValueError("%s in %r" % (exc, ip_str)) from None
ipaddress.AddressValueError: Octet 256 (> 255) not permitted in '256.1.0.2'
```

Reply:
Replaced, thanks.